eclipse-che/che-server
# chore: remove some outdated che.workspace.* properties

### What does this PR do?
Removes some che-server properties (mainly `che.workspace.*`) which are no longer used in Eclipse Che.
**Note:** I tried to remove the minimum amount of code required to remove these properties while preventing build and runtime errors. There is more code related to the removed properties could be removed if desired.

The following properties have been removed in this PR:
```
che.workspace.projects.storage 
che.workspace.logs.root_dir 
che.workspace.http_proxy
che.workspace.https_proxy
che.workspace.no_proxy
che.infra.kubernetes.workspace_start_timeout_min
che.workspace.sidecar.image_pull_policy
che.workspace.activity_check_scheduler_period_s
che.workspace.activity_check_scheduler_delay_s
che.workspace.activity_cleanup_scheduler_period_s
che.workspace.activity_cleanup_scheduler_initial_delay_s
che.workspace.pod.node_selector
che.workspace.pod.tolerations_json
che.workspace.auto_start
che.workspace.pool.type
che.workspace.pool.exact_size
che.workspace.pool.cores_multiplier
che.workspace.probe_pool_size
che.workspace.server.ping_success_threshold
che.workspace.server.ping_interval_milliseconds
```

### Screenshot/screencast of this PR
n/a


### What issues does this PR fix or reference?
Part of https://github.com/eclipse/che/issues/21411

### How to test this PR?
While working on this PR, after the removal of each property I did the following to test my changes:
**1.** Build and test relevant modules, to ensure no errors (note that in some cases, the relevant modules were not `infrastructure-kubernetes` and `infrastructure-openshift`:
```bash
mvn clean install -DskipTests=false -Dfindbugs.skip=true -Dmdep.analyze.skip=true -Dlicense.skip=true -Dgwt.compiler.localWorkers=2 -T 1C -pl ':infrastructure-kubernetes,:infrastructure-openshift' --amd -Pintegration
```

**2.** Do a full build with tests, ensure no errors
`mvn clean install -Pintegration`

**3.** Ensure it is possible to deploy che-server with chectl with no errors.
  **a.** From within the che-server repo, build the che-server image and deploy it to your quay repository:
`cd ./dockerfiles ; ./build.sh ; docker tag quay.io/eclipse/che-server:next quay.io/{QUAY_USER_ID}/che-server:next ; docker push quay.io/{QUAY_USER_ID}/che-server:next ; `

  **b.** Deploy che using your image of che-server built from this PR (I tested with Minikube)
`chectl server:deploy --installer operator -p minikube -i=quay.io/{QUAY_USER_ID}/che-server:next  --debug`

  **c.** Ensure che-server does not contain any errors in its logs:
`chectl server:logs -d ~/tmp/ && cd ~/tmp/eclipse-che/ && cd $(find . -name che.log -exec dirname {} + ) && cat che.log`



### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
